python/foglamp/services/core/api/service.py

```diff
@@ -95,57 +95,70 @@ def get_service_records():
         payload = PayloadBuilder().SELECT("name").WHERE(['name', '=', name]).payload()
         result = storage.query_tbl_with_payload('scheduled_processes', payload)
         count = result['count']
         if count != 0:
             raise web.HTTPBadRequest(reason='A service with that name already exists')
 
+        # Check that the schedule name is not already registered
+        payload = PayloadBuilder().SELECT("schedule_name").WHERE(['schedule_name', '=', name]).payload()
+        result = storage.query_tbl_with_payload('schedules', payload)
+        count = result['count']
+        if count != 0:
+            raise web.HTTPBadRequest(reason='A schedule with that name already exists')
+
         # First create the scheduled process entry for our new service
         if service_type == 'south':
             script = '["services/south"]'
+            plugin_module_path = "foglamp.plugins.south"
         if service_type == 'north':
             script = '["services/north"]'
+            plugin_module_path = "foglamp.plugins.north"
         payload = PayloadBuilder().INSERT(name=name, script=script).payload()
         try:
             res = storage.insert_into_tbl("scheduled_processes", payload)
         except Exception as ins_ex:
             raise web.HTTPInternalServerError(reason='Failed to created scheduled process. {}'.format(str(ins_ex)))
 
-        # Now create a configuration category with the minimum to load the plugin
-        # TODO It would be better to load the default configuration from the plugin
-        # and use this, however we should extract the plugin loading code so that is shared
-        new_category = {"plugin":
-                            {"type": "string",
-                             "default": plugin,
-                             "description": "Python module name of the plugin to load"}
-                       }
-        category_desc = '{} service configuration'.format(name)
-        config_mgr = ConfigurationManager(storage)
-        await config_mgr.create_category(category_name=name, category_description=category_desc,
-                                     category_value=new_category, keep_original_items=True)
-
-        # Check that the process is not already registered
-        payload = PayloadBuilder().SELECT("schedule_name").WHERE(['schedule_name', '=', name]).payload()
-        result = storage.query_tbl_with_payload('schedules', payload)
-        count = result['count']
-        if count != 0:
-            raise web.HTTPBadRequest(reason='A schedule with that name already exists')
-
-        # Finally add a schedule to run the new service at startup
-        schedule = StartUpSchedule()
+        # Now load the plugin to fetch its configuration
+        try:
+            try:
+                import_file_name = "{path}.{dir}.{file}".format(path=plugin_module_path, dir=plugin, file=plugin)
+                _plugin = __import__(import_file_name, fromlist=[''])
+            except Exception as ex:
```

Comment:
instead of having nested try/except, please capture this in `except ImportError:` and map to HTTP bad request or not found as appropriate.

Also please add a comment for path=plugin_module_path dir=plugin, file=plugin in docstring or atleast just above the `import_file_name`